text sensor filter documentation (#1497)
OttoWinter/esphomedocs: components/text_sensor/index.rst

- **on_value** (*Optional*, :ref:`Automation <automation>`): An automation to perform
  when a new value is published. See :ref:`text_sensor-on_value`.
- **on_raw_value** (*Optional*, :ref:`Automation <automation>`): An automation to perform
  when a new value is received that hasn't passed through any filters. See :ref:`text_sensor-on_raw_value`.

.. _text_sensor-filters:

Text Sensor Filters
-------------------

ESPHome allows you to do some basic pre-processing of
text_sensor values before they’re sent to Home Assistant. This is for example

> useful if you want to apply manipulate the text_sensor string in some fashion.

Reviewer: ```suggestion
useful if you want to manipulate the text_sensor string in some fashion.
```

There are a lot of filters that sensors support. You define them by adding a ``filters``
block in the text_sensor configuration (at the same level as ``platform``; or inside each text_sensor block

> for platforms with multiple sensors)

Reviewer: ```suggestion
for platforms with multiple sensors).
```

Filters are processed in the order they are defined in your configuration.

.. code-block:: yaml

    # Example filters:
    filters:
      - to_upper:
      - to_lower:
      - append: "_prefix"
      - prepend: "suffix_"
      - substitute:
        - "suf -> foo"
        - "pre -> bar"
      - lambda: return {"Hello World"};

``to_upper``
************

> Converts all characters within a string to uppercase.

Reviewer: Might be nice to note that we only support a-z, something like this:
```suggestion
Converts all characters within a string to uppercase (only the English alphabet is supported at this time).
```

.. code-block:: yaml

    # Example configuration entry
    - platform: template
      # ...
      filters:
        - to_upper:

``to_lower``
************

> Converts all characters within a string to lowercase.

Reviewer: ```suggestion
Converts all characters within a string to lowercase (only the English alphabet is supported at this time).
```

.. code-block:: yaml

    # Example configuration entry
    - platform: template
      # ...
      filters:
        - to_lower:

``append``
**********

Adds a string to the end of the current string.

.. code-block:: yaml

    # Example configuration entry
    - platform: template
      # ...
      filters:
        - append: "_prefix"

``prepend``
***********

Adds a string to the start of the current string.

.. code-block:: yaml

    # Example configuration entry
    - platform: template
      # ...
      filters:
        - prepend: "suffix_"

``substitute``
**************

> Search the text sensor current string for a string, and replace it with a second string.

Reviewer: ```suggestion
Search the current value of the text sensor for a string, and replace it with another string.
```
